test/ng/compileSpec.js

```diff
@@ -7815,4 +7817,125 @@ describe('$compile', function() {
         expect(element.hasClass('fire')).toBe(true);
       }));
   });
+
+  describe('element replacement', function() {
+    it('should broadcast $destroy only on removed elements, not replaced', function() {
```

Comment:
Removing `jqLite(firstElementToRemove).off('$destroy');` doesn't cause anything to fail. Even this test designed to fail in that case isn't failing!  :worried:

I thought of another case though which I think might show that not invoking the DOM $destory is (sometimes) wrong...

```html
<div ng-form="myForm" lower-priority-replace-directive>
```

In this case the `ng-form` directive runs first which adds itself to the parent form then does `formElement.on('$destroy', function() { parentFormCtrl.$removeControl(controller); ... })`. Then the other lower priority directive replaces that element with something brand new, that replaced element is not removed from the ng-form because no $destroy was fired. The replaced ng-form controller left behind a) is leaked and b) might cause form validation issues?

However if that low priority directive was doing transclude instead of replace then maybe not invoking the DOM $destory is valid? Meaning `replaceWith` would need an `invokeDestroy` argument...